File: Algorithms/Дынамічнае праграмаваньне/Problem_06. Строга нарастальная без парываў падпасьлядоўнасьць/source.py
def LIS_l(s, arr):

    l = 1
    t = [0 for i in range(s + 1)]
    t[0] = arr[0]

    for i in range(1, s):

        if (arr[i] < t[0]): t[0] = arr[i]
        elif (t[l-1] < arr[i]):
            t[l] = arr[i]
            l = l + 1
        else: t[indextn(t, -1, l-1, arr[i])] = arr[i]

    return l

def indextn(arr, l, r, key):

    while (1 < r - l):
        mdm = l + (r - l)//2
        if (key <= arr[mdm]): r = mdm
        else: l = mdm
    return r

with open('input.txt') as f:
    w = [int(x) for x in next(f).split()]
    h = [int(x) for x in next(f).split()]
    answer = LIS_l(w[0], h)
    z = open("output.txt","w")
    z.write(str(answer))
    z.close()
    f.close()

File: Algorithms/Структуры дадзеных ды графы/Problem_0.04. Пабудаваць матрыцу сумежнасьці/source.cpp
#include <fstream>
#include <iostream>
#include <vector>

int main() {

    std::ifstream in("input.txt"); std::ofstream out("output.txt");

    int n, m; in >> n >> m;
    std::vector <std::vector <int> > A(n);
    for (auto &i:A) i = std::vector <int> (n);

    for (int k = 0; k < m; ++k) {
        
        int i; in >> i; i--;
        int j; in >> j; j--;
        
        A[i][j] = 1;
        A[j][i] = 1;
        
    }
    
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            out << A[i][j] << " ";
        }
        out << std::endl;
    }
    
    in.close(); out.close();
    return 0;
}
